ingestor/adx/tasks.go

```diff
@@ -215,16 +218,84 @@ func (t *ManagementCommandTask) Run(ctx context.Context) error {
 		stmt := kql.New(".execute script<|").AddUnsafe(command.Spec.Body)
 		if _, err := t.kustoCli.Mgmt(ctx, stmt); err != nil {
 			logger.Errorf("Failed to execute management command %s.%s: %v", command.Spec.Database, command.Name, err)
-			if err = t.store.UpdateStatus(ctx, command, err); err != nil {
+			if err = t.store.UpdateStatus(ctx, &command, err); err != nil {
 				logger.Errorf("Failed to update management command status: %v", err)
 			}
 		}
 
 		logger.Infof("Successfully executed management command %s.%s", command.Spec.Database, command.Name)
-		if err := t.store.UpdateStatus(ctx, command, nil); err != nil {
+		if err := t.store.UpdateStatus(ctx, &command, nil); err != nil {
 			logger.Errorf("Failed to update success status: %v", err)
 		}
 	}
 
 	return nil
 }
+
+type SummaryRuleTask struct {
+	store    storage.CRDHandler
+	kustoCli StatementExecutor
+}
+
+func NewSummaryRuleTask(store storage.CRDHandler, kustoCli StatementExecutor) *SummaryRuleTask {
+	return &SummaryRuleTask{
+		store:    store,
+		kustoCli: kustoCli,
+	}
+}
+func (t *SummaryRuleTask) Run(ctx context.Context) error {
+	summaryRules := &v1.SummaryRuleList{}
+	if err := t.store.List(ctx, summaryRules); err != nil {
+		return fmt.Errorf("failed to list summary rules: %w", err)
+	}
+	for _, rule := range summaryRules.Items {
+		if rule.Spec.Database != t.kustoCli.Database() {
+			continue
+		}
+		if rule.DeletionTimestamp != nil {
+			// rule is being deleted, we might possibly want to delete the associated
+			// destination table, but there is a potential for unintended historical
+			// data loss if the user doesn't understand this consequence.
+			continue
+		}
+
+		cnd := rule.GetCondition()
+		if cnd != nil && cnd.Status == metav1.ConditionTrue && time.Since(cnd.LastTransitionTime.Time) < rule.Spec.Interval.Duration {
+			if logger.IsDebug() {
+				logger.Debugf("Skipping summary rule %s.%s as it was executed recently %s", rule.Spec.Database, rule.Name, cnd.LastTransitionTime)
+			}
+			continue
+		}
+
+		if cnd == nil {
+			cnd = &metav1.Condition{}
+		}
+
+		// Execution occurs asynchronously, so we don't have to concern ourselves with blocking
+		startTime := time.Now().Add(-rule.Spec.Interval.Duration)
```

Comment:
We need to make these start and end time fall on regular intervals.  I think we should have a minimum of minutely boundaries so we don't allow running something every few seconds faster.  We'd need the start and end times to zero out all the second, millis, nanos.